core/capabilities/remote/trigger_subscriber.go

```diff
@@ -273,3 +305,47 @@ func (s *triggerSubscriber) HealthReport() map[string]error {
 func (s *triggerSubscriber) Name() string {
 	return s.lggr.Name()
 }
+
+// SetConfig sets the remote trigger configuration, capability info, and DON information dynamically
+func (s *triggerSubscriber) SetConfig(config *commoncap.RemoteTriggerConfig, capInfo commoncap.CapabilityInfo, localDONID uint32, remoteDON commoncap.DON, aggregator types.Aggregator) error {
+	if config == nil {
+		return errors.New("no config provided")
+	}
+	if capInfo.ID == "" || capInfo.ID != s.capabilityID {
+		return fmt.Errorf("capability info provided does not match the subscriber's capabilityID: %s != %s", capInfo.ID, s.capabilityID)
+	}
+	if localDONID == 0 {
+		return errors.New("localDONID=0 provided")
+	}
+	if remoteDON.ID == 0 || len(remoteDON.Members) == 0 {
+		return errors.New("empty remoteDON provided")
+	}
+	if aggregator == nil {
+		return errors.New("aggregator not set - call SetAggregator() before SetConfig()")
+	}
+	config.ApplyDefaults()
+	// Rebuild the capDonMembers map
+	capDonMembers := make(map[p2ptypes.PeerID]struct{})
+	for _, member := range remoteDON.Members {
+		capDonMembers[member] = struct{}{}
+	}
+
+	s.cfgMu.Lock()
+	defer s.cfgMu.Unlock()
+	// always replace the whole dynamicConfig object to avoid inconsistent state
+	s.cfg = dynamicConfig{
+		remoteConfig:  config,
+		capInfo:       capInfo,
+		capDonInfo:    remoteDON,
+		capDonMembers: capDonMembers,
+		localDonID:    localDONID,
+		aggregator:    aggregator,
+	}
+	return nil
+}
+
+func (s *triggerSubscriber) getConfigLocked() dynamicConfig {
```

Comment:
Maybe use an atomic.Pointer for cfg instead?  Semantically just as ugly, but prevents any future maintainer accidentally using the cfg object directly without going through a mem sync.